platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -13,204 +13,259 @@
  */
-public class ConfigurationInstaller implements SynchronousConfigurationListener {
+public class ConfigurationInstaller implements ConfigurationPersistencePlugin {
   private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
 
-  private final ConfigurationAdmin configurationAdmin;
+  private static final Logger SECURITY_LOGGER = LoggerFactory.getLogger("securityLogger");
+
+  private final ConfigurationAdmin ddfConfigAdmin;
 
-  private final ConfiguratorFactory configuratorFactory;
+  private final List<PersistenceStrategy> strategies;
 
-  private final ConfigActions configActions;
+  private final EncryptionService encryptionService;
 
-  private final Map<String, File> pidFileMap;
+  private final Map<String, CachedConfigData> pidDataMap;
 
+  private final ConfigurationContextFactory configurationContextFactory;
+
+  /**
+   * {@code SortedServiceList} implements {@link java.util.List} using generic parameter <code>
+   * &lt;T&gt;</code> and not a concrete type, so the ctor cannot be matched by the blueprint
+   * container if we did <code>List&lt;PersistenceStrategy&gt;</code>
+   *
+   * <p>See https://issues.apache.org/jira/browse/ARIES-960
+   */
   public ConfigurationInstaller(
-      ConfigurationAdmin configurationAdmin,
-      ConfiguratorFactory configuratorFactory,
-      ConfigActions configActions) {
-    this.configurationAdmin = configurationAdmin;
-    this.pidFileMap = new ConcurrentHashMap<>();
-
-    this.configuratorFactory = configuratorFactory;
-    this.configActions = configActions;
+      ConfigurationAdmin ddfConfigAdmin,
+      List strategies,
+      EncryptionService encryptionService,
+      ConfigurationContextFactory configurationContextFactory) {
+    this.ddfConfigAdmin = ddfConfigAdmin;
+    this.strategies = strategies;
+    this.encryptionService = encryptionService;
+    this.pidDataMap = new ConcurrentHashMap<>();
+
+    this.configurationContextFactory = configurationContextFactory;
   }
 
   /** @return a read-only map of configuration pids to felix file install properties. */
-  Map<String, File> getPidFileMap() {
-    return Collections.unmodifiableMap(pidFileMap);
+  Map<String, CachedConfigData> getPidDataMap() {
+    return Collections.unmodifiableMap(pidDataMap);
   }
 
+  /**
+   * @see ConfigurationPersistencePlugin
+   * @throws IOException if an error occurs reading configuration data
+   * @throws InvalidSyntaxException technically impossible since {@code null} is being passed as the
+   *     configuration filter. See {@link ConfigurationAdmin#listConfigurations(String)}.
+   */
   public void init() throws IOException, InvalidSyntaxException {
-    Configuration[] configs = configurationAdmin.listConfigurations(null);
+    Configuration[] configs = ddfConfigAdmin.listConfigurations(null);
     if (configs == null) {
       return;
     }
-    pidFileMap.putAll(
+    pidDataMap.putAll(
```

Comment:
I'm pretty sure osgi guarantees that if our init method gets called, we are already available to be called. That being said, since we are integrating with config admin by force, we might miss a config if we come online precisely after we are invoked but before the config is officially created and can be queried. 

❓ One option is to also implement config listener like before, but use it only as a validation step against our map.

Reply:
@Lambeaux My concern is that when we do come up, there might already be configs in the system that we missed and will never receive events for. I am suggesting simply adding a query back to config admin to get the list of all config objects after we are initialized and have added the listener (or it being added automatically). Worst case scenario, you will get identical objects form events you might receive right after the listener is added but best case scenario, you will get those that you haven't got any as they were already there.

Reply:
@paouelle's concern is valid. Sounds like there may have to be a separate "initialize" method that the DelegatingPersistenceManager calls on the first interaction with the plugin. Or that logic could be pushed down into the plugin itself. Either solution is a little awkward, in my opinion.